### ISSUE TITLE
Question about implementation of loss function

### ISSUE THREAD

**Issue by tunglm2203** [brown-palm/GCPC#1]
Hi, thanks for sharing the code!

In the paper, it mentioned that the training objective of the TrajNet is MSE between the reconstructed and the ground-truth trajectory. However, in the **antmaze** environment, you only use the first 2 dimensions to calculate MSE. Could you explain this difference?

https://github.com/brown-palm/GCPC/blob/c75e65d1908c4cf48bb12ae53d2b1999e0af9010/gcpc/model/trajnet.py#L173-L179

**Comment by zilaiz** [comment]
Hi, thank you for your interest in our work, and sorry for the delayed response. For goal (state)-conditioned tasks, where the goal could be defined in a subspace of the original state space (e.g. Antmaze), we observed that calculating MSE on the goal space allows us to extract good enough representations (i.e. employ GCPC to capture future waypoints) from state-only trajectories. This can potentially help reduce the computational cost and enable optimization for faster training and inference. However, in more general cases where the task may not always be goal-conditioned, by default we calculate MSE over the full state space between the reconstructed and the ground-truth trajectory.
